library/Makefile

```diff
@@ -1,8 +1,21 @@
 
 # Also see "include/mbedtls/config.h"
 
+ifeq ($(CC),iccarm)
+CFLAGS	?= --arm -Om --silent
+
+# gcc comparable warning flags are not present in IAR.
+# All warnings are enabled by default.
+WARNING_CFLAGS ?=
+else
 CFLAGS	?= -O2
 WARNING_CFLAGS ?=  -Wall -W -Wdeclaration-after-statement
+endif
+
+# IAR archive command does not have ar compatible options 
+ifneq ($(AR),iarchive)
+AR += -rcs
```

Comment:
While this is allowed, I do not think adding flags to the `AR` variable is a good practice. The GNU Make documentation [here](https://www.gnu.org/software/make/manual/html_node/Implicit-Variables.html) says:
>AR
>Archive-maintaining program; default ‘ar’.

If you would like to pass flags to the archiver, I suggest using the variable `ARFLAGS`.